modules/private.php

```diff
@@ -0,0 +1,265 @@
+ *
+ * @param int   $blog Current WordPress blod id.
+ * @param Mixed $user Current WordPress user or user_id.
+ */
+function is_private_blog_user( $blog, $user ) {
+	global $wpdb;
+
+	if ( ! is_object( $user ) ) {
+		$user = new WP_User( $user );
+	}
+
+	if ( ! $user->ID ) {
+		return false;
+	}
+
+	if ( is_numeric( $blog ) ) {
+		$blog_id = intval( $blog );
+	} elseif ( is_object( $blog ) ) {
+		$blog_id = $blog->blog_id;
+	} elseif ( is_string( $blog ) ) {
+		$blog    = get_blog_info( $blog, '/', 1 );
+		$blog_id = $blog->blog_id;
+	} else {
+		$blog_id = $wpdb->blogid;
+	}
+
+	// check if the user has read permissions.
+	$the_user = wp_clone( $user );
+	$the_user->for_blog( $blog_id );
+	return $the_user->has_cap( 'read' );
+}
+
+/**
+ * Tests whether the current blog is private and not spam/suspended/deleted.
+ *
+ * @param int $_blog_id Current WordPress blod id.
+ */
+function is_private_blog( $_blog_id = null ) {
+	global $blog_id;
+
+	if ( empty( $_blog_id ) ) {
+		$_blog_id = $blog_id;
+	}
+
+	$blog_details = get_blog_details( $_blog_id );
+
+	return ( ( '-1' === $blog_details->public ) &&
+		( ! isset( $blog_details->deleted ) || ! $blog_details->deleted ) &&
+		( ! isset( $blog_details->archived ) || ! $blog_details->archived ) &&
+		( ! isset( $blog_details->spam ) || ! $blog_details->spam )
+	);
+}
+
+/**
+ * Filters new comments so that users can't comment on private blogs
+ *
+ * @param array $comment Documented in wp-includes/comment.php.
+ */
+function privatize_blog_comments( $comment ) {
+	privatize_blog( null );
+	return $comment;
+}
+
+/**
+ * Extend the 'Site Visibility' privacy options to also include a private option
+ */
+function privatize_blog_priv_selector() {
+	?>
+	<br /><input id="blog-private" type="radio" name="blog_public" value="-1" <?php checked( '-1', get_option( 'blog_public' ) ); ?> />
+	<label for="blog-private"><?php esc_html_e( 'I would like my site to be private, visible only to myself and users I choose', 'jetpack' ); ?></label>
+	<?php
+}
+
+/**
+ * Hides the blog's name on the login form for private blogs.
+ */
+function privatize_blog_maybe_mask_blog_name() {
+	if ( ! is_private_blog() ) {
+		return;
+	}
+
+	add_filter( 'bloginfo', 'privatize_blog_mask_blog_name', 3, 2 );
+}
+
+/**
+ * Replaces the the blog's "name" value with "Protected Blog"
+ *
+ * @see privatize_blog_maybe_mask_blog_name()
+ * @param mixed $value The requested non-URL site information.
+ * @param mixed $what  Type of information requested.
+ */
+function privatize_blog_mask_blog_name( $value, $what ) {
+	if ( in_array( $what, array( 'name', 'title' ), true ) ) {
+		$value = __( 'Protected Blog', 'jetpack' );
+	}
+
+	return $value;
+}
+
+/**
+ * Changes a blog to private
+ */
+function make_blog_private() {
+	update_option( 'blog_public', -1 );
+}
+
+/**
+ * Changes a blog to public
+ */
+function make_blog_public() {
+	update_option( 'blog_public', 1 );
+}
+
+add_action( 'parse_request', 'privatize_blog', 100 );
```

Comment:
Is this enough to make sure we block access to RSS feeds, Core's REST API, and WordPress.com's REST API requests for that site?

Reply:
As long as the Jetpack site is connected to WordPress.com and the JSON API module is active in Jetpack (it's enabled by default and there is no easy way to deactivate it, the UI is hidden), the site's content will be available via the WordPress.com REST API. Here is an example:

https://public-api.wordpress.com/rest/v1.1/sites/53262711/posts/

Reply:
It looks like they are already blocked in the WPCOM REST API: https://opengrok.a8c.com/source/xref/trunk/public.api/rest/class.json-api-endpoints.php#1121